wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java

```diff
@@ -645,30 +658,63 @@ public void test_pooledConnectionFailover() throws SQLException, InterruptedExce
 
   @TestTemplate
   @EnableOnTestFeature(TestEnvironmentFeatures.FAILOVER_SUPPORTED)
-  public void test_pooledConnectionFailoverWithClusterURL() throws SQLException, InterruptedException {
-    Properties props = getPropsWithFailover();
+  public void test_pooledConnectionFailoverWithClusterUrl() throws SQLException, InterruptedException {
+    System.setProperty(HIKARI_VALIDATION_TIMEOUT_PROPERTY,
+        String.valueOf(TimeUnit.MINUTES.toMillis(10)));
 
+    final Properties props = getDefaultPropsNoPlugins();
+    PropertyDefinition.PLUGINS.set(props, "readWriteSplitting,failover");
+
+    final int maxPoolSize = 3;
     final HikariPooledConnectionProvider provider =
-        new HikariPooledConnectionProvider(getHikariConfig(1));
+        new HikariPooledConnectionProvider(getHikariConfig(maxPoolSize));
     ConnectionProviderManager.setConnectionProvider(provider);
 
     final Connection initialWriterConn;
     final Connection newWriterConn;
+    final String originalWriterId;
+    final String nextWriterId;
+
+    List<Connection> setupConnections = new ArrayList<>();
+    for (int i = 0; i < maxPoolSize; i++) {
+      Connection conn = DriverManager.getConnection(ConnectionStringHelper.getWrapperClusterEndpointUrl(), props);
```

Comment:
Cannot use try-with-resources for the setup connections because then there would only be one connection in the pool (the same connection would be returned each time getConnection is called). Doing it this way instead allows the pool to be filled before closing the connections